spdif_rx.c

```diff
@@ -494,7 +533,7 @@ static int _spdif_rx_analyze_capture(spdif_rx_samp_freq_t* samp_freq, bool* inve
         int bit_pos = 0; // position within current dword
         int word_idx = 0;
         while (true) {
-            int r = __builtin_clz((cur) ? ~shift_reg : shift_reg); // leading zeros
+            int r = _spdif_count_leading_zeros((cur) ? ~shift_reg : shift_reg); // leading zeros
```

Comment:
How about simply rescuing for the case of zero input? (not to define _spdif_count_leading_zeros())

```suggestion
            uint32_t pol_shift_reg = cur ? ~shift_reg : shift_reg;
            int r = pol_shift_reg == 0 ? 32 : __builtin_clz(pol_shift_reg); // leading zeros (avoid the case for zero as undefined)
```

Reply:
For some reason this doesn't work, the compiler still removes the code. Here is a godbolt link that can be used for testing: https://godbolt.org/z/zvzM8Te1f

Reply:
Okay, thanks. at least the code works in my environment with `arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1`. Let me try to investigate. Going back to c function looks to me not good idea for the moment.

Reply:
Thanks for the information.
* I'm not for sure if none-eabi version of ARM GCC is not used actually in Arduino environment, It looks to me that ARM GCC 14.2.0 (unknown-eabi) version should be used to designate the target CPU.
* If ARM GCC 14.2.0 (unknown-eabi) version is used, when the independent function is used for wrapping `__builtin_clz()` then it looks working fine with `-O3`. (If function is attached with `inline` attribute, it fails.)
* Thus, for `-mcpu=cortex-m0`, which is targeting to rp2040 (Pico), `bl      __clzsi2` instruction remains. For `-mcpu=cortex-m4`, which is targeting to rp2350 (Pico2), `clz` instruction remains. They are expected compiled results.

https://godbolt.org/z/zxfT3P615

Reply:
Precisely, CPU target is `m0plus` for rp2040.
https://godbolt.org/z/h5fz79bWh